Doc/library/io.rst

```diff
@@ -106,6 +106,54 @@ stream by opening a file in binary mode with buffering disabled::
 The raw stream API is described in detail in the docs of :class:`RawIOBase`.
 
 
+.. _io-text-encoding:
+
+Text Encoding
+-------------
+
+The default encoding of :class:`TextIOWrapper` and :func:`open` is
+locale-specific (:func:`locale.getpreferredencoding(False) <locale.getpreferredencoding>`).
+
+But many developers forget to specify encoding when opening text files
+encoded in UTF-8 (e.g. JSON, TOML, Markdown, etc...) since most Unix
+platforms uses UTF-8 locale by default. It cause a bug because locale-specific
+encoding is not UTF-8 for most Windows users. For example::
+
+   # may not work on Windows when non-ASCII characters in the file.
+   with open("README.md") as f:
+       long_description = f.read()
+
+Additionally, Python may change the default text encoding to UTF-8 in the
+future, although there is no plan yet.
+
+So it is highly recommended to specify encoding explicitly when opening text
+files. If you want to use UTF-8, specify ``encoding="utf-8"``. If you need to
+use locale-specific encoding, ``encoding="locale"`` is supported since Python
+3.10.
+
+When you need to run code using the default encoding to open UTF-8 files on
+Windows, you can enable the UTF-8 mode. See also the
+:ref:`UTF-8 mode on Windows <win-utf8-mode>`
```

Comment:
```suggestion
When you need to run existing code on Windows that attempts to opens
UTF-8 files using the default locale encoding, you can enable the UTF-8
mode. See :ref:`UTF-8 mode on Windows <win-utf8-mode>`.
```

It wasn't fully clear what the intended use case for this was over just modifying the code, so I took my best guess, as well as fixed more grammar issues, clarified the prose and avoided ambiguous phrasing/structure.